### PR TITLE
fix(core): remove code to treat <template> elements as <ng-template>

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: packages/compiler/src/template_parser/template_parser.ts [hunk 1]
```diff
@@ -912,16 +914,12 @@ function isEmptyExpression(ast: AST): boolean {
...
+    el: html.Element, reportDeprecation: (m: string, span: ParseSourceSpan) => void): boolean {
   if (isNgTemplate(el.name)) return true;
   const tagNoNs = splitNsName(el.name)[1];
   // `<template>` is HTML and case insensitive
   if (tagNoNs.toLowerCase() === TEMPLATE_ELEMENT) {
-    if (enableLegacyTemplate && tagNoNs.toLowerCase() === TEMPLATE_ELEMENT) {
-      reportDeprecation(TEMPLATE_ELEMENT_DEPRECATION_WARNING, el.sourceSpan !);
-      return true;
-    }
+    throw new Error(TEMPLATE_ELEMENT_DEPRECATION_WARNING);
```
Comment (JoostK) [inline packages/compiler/src/template_parser/template_parser.ts]: Throwing an error doesn't allow for [HTML5's template element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) to work, that seems a bit too much doesn't it? ;)

File: packages/compiler/src/template_parser/template_parser.ts [hunk 2]
```diff
@@ -912,16 +914,12 @@ function isEmptyExpression(ast: AST): boolean {
 
 // `template` is deprecated in 4.x
 function isTemplate(
-    el: html.Element, enableLegacyTemplate: boolean,
-    reportDeprecation: (m: string, span: ParseSourceSpan) => void): boolean {
+    el: html.Element, reportDeprecation: (m: string, span: ParseSourceSpan) => void): boolean {
```
Comment (JoostK) [inline packages/compiler/src/template_parser/template_parser.ts]: We may also get rid of the `reportDeprecation` parameter here.

**Comment by matsko** [comment]
Seems this guy beat it to me: :) https://github.com/angular/angular/commit/0ebd577db456ee4660a61828c6367bc10dcddb9c

**Comment by matsko** [comment]
Sorry @JoostK that I didn't see your comments. This was just a ploy to get @vicb back into coding this fix :)

**Comment by JoostK** [comment]
@matsko hahaha nicely played 😄

**Pull request closed without merging**